src/ClientData/include/ClientData/ScopeIdProvider.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CLIENT_DATA_API_SCOPE_ID_PROVIDER_H_
+#define ORBIT_CLIENT_DATA_API_SCOPE_ID_PROVIDER_H_
+
+#include <cstdint>
+
+#include "ClientData/TimerTrackDataIdManager.h"
+#include "GrpcProtos/capture.pb.h"
+
+namespace orbit_client_data {
+
+// The inferface defines a map from `TimerInfo` to ids. When called twice on identical `TimerInfo`
+// instances, it returns the same ids.
```

Comment:
Maybe state that the ids are used for grouping (e.g. on the live tab). You could also add the go-link for the manual instrumenation grouping document.